test/functional/pdo_sqlsrv/pdo_ae_output_param_binary_size.phpt

```diff
@@ -0,0 +1,211 @@
+                dropProc($conn, $spname);
+                $spSql = "CREATE PROCEDURE $spname (
+                                @c_det $type OUTPUT, @c_rand $type OUTPUT ) AS
+                                SELECT @c_det = c_det, @c_rand = c_rand FROM $tbname";
+                $conn->query($spSql);
+                
+                // call stored  procedure
+                $outSql = getCallProcSqlPlaceholders($spname, 2);
+                foreach ($pdoParamTypes as $pdoParamType) {
+                    $stmt = $conn->prepare($outSql);
+                    trace("\nParam $pdoParamType with INOUT = $inout\n");
+                    
+                    if ($inout && $pdoParamType != PDO::PARAM_STR) {
+                        // Currently do not support getting binary as strings + INOUT param
+                        $paramType = $pdoParamType | PDO::PARAM_INPUT_OUTPUT;
+                    } else {
+                        $paramType = $pdoParamType;
+                    }
+
+                    $det = "";
+                    $rand = "";
+                    if ($pdoParamType == PDO::PARAM_STR) {
+                        $stmt->bindParam(1, $det, $paramType, $length, PDO::SQLSRV_ENCODING_BINARY); 
+                        $stmt->bindParam(2, $rand, $paramType, $length, PDO::SQLSRV_ENCODING_BINARY); 
+                    } elseif ($pdoParamType == PDO::PARAM_BOOL || $pdoParamType == PDO::PARAM_INT) {
+                        $det = $rand = 0;
+                        $stmt->bindParam(1, $det, $paramType, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+                        $stmt->bindParam(2, $rand, $paramType, PDO::SQLSRV_PARAM_OUT_DEFAULT_SIZE);
+                    } else {
+                        $stmt->bindParam(1, $det, $paramType, $length); 
+                        $stmt->bindParam(2, $rand, $paramType, $length);
+                    }
+                    
+                    try {
+                        $stmt->execute();
+
+                        $errMsg = "****$dataType as $pdoParamType failed with INOUT = $inout:****\n";
+                        if ($pdoParamType == PDO::PARAM_STR) {
+                            if ($det !== $input0 || $rand !== $input1) {
+                                printValues($errMsg, $det, $rand, $input0, $input1);
+                            }
+                        } elseif ($pdoParamType == PDO::PARAM_BOOL) {
+                            // for boolean values, they should all be bool(true)
+                            // because all floats are non-zeroes
+                            // This only occurs without AE
+                            // With AE enabled, this would have caused an exception
+                            if (!$det || !$rand) {
+                                printValues($errMsg, $det, $rand, $input0, $input1);
+                            }
+                        } else {
+                            // $pdoParamType is PDO::PARAM_INT

```

Comment:
what about PDO::PARAM_LOB and PDO::PARAM_NULL?

Reply:
they would have thrown exceptions